core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -698,18 +702,16 @@ public boolean process()
                 return false;
             }
 
-            int[] combinationIdToPosition = new int[maxCardinality];
-            Arrays.fill(combinationIdToPosition, -1);
-            calculateCombinationIdsToPositionMapping(combinationIdToPosition);
-
             // putIfAbsent will rehash automatically if rehash is needed, unless there isn't enough memory to do so.
             // Therefore needRehash will not generally return true even if we have just crossed the capacity boundary.
-            for (int i = 0; i < maxCardinality; i++) {
-                if (needRehash()) {
-                    return false;
-                }
-                if (combinationIdToPosition[i] != -1) {
-                    putIfAbsent(combinationIdToPosition[i], page);
+            for (int combinationIndex = nextCombinationIndex; combinationIndex < combinationIdToPosition.length; combinationIndex++) {
```

Comment:
`combinationIndex` -> `combinationId`

Reply:
See above rationale, happy to change it if you insist.